back/engines/commercial/analysis/app/lib/analysis/auto_tagging_method/nlp_topic.rb

```diff
@@ -50,5 +55,18 @@ def parse_topic_modeling_response(response)
     def max_topics(inputs_count)
       [inputs_count, (Math.log(inputs_count, 5) * 6).ceil].min
     end
+
+    def fit_inputs_in_context_window(inputs, project_title)
+      prompt = inputs_prompt(inputs, project_title)
+      tokens_for_response = TOKENS_PER_TOPIC * max_topics(inputs.size)
+      tokens = LLM::AzureOpenAI.token_count(prompt) + tokens_for_response
```

Comment:
Quick question bf doing a complete review: why do we have to take `tokens_for_response` into account here? As far as I can tell, topics are not included in the `topic_modeling` prompt, or am I missing something?

Reply:
@adessy No, they are not included in the prompt but the context window limit is expressed on the size of the prompt plus the size of the response returned. I we include just a slightly smaller amount of tokens than allowed in the prompt, we risk that the response would be cut off.